### DISCUSSION TITLE
Error "Size of velocity boundaryTypeMap does not match number of boundary IDs in mesh!".

### DISCUSSION THREAD

**Discussion by isurug1997** [Nek5000/nekRS#523] [Compiling]
Dear Neks,
I am trying to run a custom periodic turbulent pipe case with a custom mesh on Nek. Once I convert the .msh file from GMSH to .re2 (following the documentation file) and try running nek, it gives me the following error.

"Error in nrsSetup: Size of velocity boundaryTypeMap does not match number of boundary IDs in mesh!".

It might be a simple fix, but I would really appreciate it if someone who might have faced this issue previously could shed some light on a way around this. 

Yours sincerely,
Isuru Gunaratne.

**Comment by stgeke** [comment]
The number of entries in `boundaryTypeMap` has to match the non-periodic `Physical Surface` defined in GMSH.

**Reply by isurug1997** [reply]
Thank you so much for your response.
In my GMSH file, I have three Physical Surfaces defined (inlet, outlet, wall & flowdomain respectively). The inlet and the outlet are periodic (defined as a periodic boundary pair upon converting the .msh file to the .re2 file). Does it mean that in the .par file, I have to input "boundaryTypeMap = wall, flowdomain" respectively to boundaryTypeMap under the [VELOCITY] section to correctly define the non-periodic Physical Surfaces?

**Reply by stgeke** [reply]
The mesh converter will translate the GMESH `Physical Surface` into nek `boundaryIDs`. In your case there should only be one (periodic faces are not real boundaries), so you want to use e.g. `boundaryTypeMap = wall`

**Reply by isurug1997** [reply]
Thank you so much for your quick response. 

This was the initial setting that was set in the .par file when I started to get this error (boundaryTypeMap = wall). I have been using the example case files (.usr, .udf, .par, .oudf, etc.) of the case "turbPipePeriodic" provided in nekRS and the only change I made was replacing the .re2 file with my mesh. 
As I thought it could be an issue with the NEK tool (gmsh2nek), I just updated my NEK5000 files from GitHub, converted the given .geo file of the example (turbPipe.geo) to .re2, and ran the example case. I am still getting the same error that I have been getting from my custom mesh case. There are no errors when I am running the .re2 file provided in the example case directly (turbPipe.re2).

My apologies for my extended response.

**Reply by stgeke** [reply]
My bad, you have to use `boundaryTypeMap = none, none, wall` (assuming boundary 1 and 2 are periodic).

**Reply by isurug1997** [reply]
Again, thank you so much for your swift correspondence.

I apologize, but despite the above correction to the .par file, the error still persists. Yes, I have set boundary 1 (inlet) and boundary 2 (outlet) as the periodic boundary pair. 

I really appreciate your assistance so far with this.

**Comment by ztdepztdep** [comment]
> Dear Neks, I am trying to run a custom periodic turbulent pipe case with a custom mesh on Nek. Once I convert the .msh file from GMSH to .re2 (following the documentation file) and try running nek, it gives me the following error.
> 
> "Error in nrsSetup: Size of velocity boundaryTypeMap does not match number of boundary IDs in mesh!".
> 
> It might be a simple fix, but I would really appreciate it if someone who might have faced this issue previously could shed some light on a way around this.
> 
> Yours sincerely, Isuru Gunaratne.

i found the solution. just set it bid in the userdat2 function. 
      do e=1,nelt
      do f=1,2*ndim
        if (cbc(f,e,1) .eq. 'W  ') boundaryID(f,e) = 1
      enddo
      enddo
